data/tutorials/tt_02_file_manipulation.md

```diff
@@ -9,123 +9,618 @@ date: 2021-05-27T21:07:30-00:00
+
+### Listing Directory Contents
+
+The `Sys.readdir` function is used to list the contents of a directory. The
+function returns an array of filenames in the directory.
+
+```ocaml
+let files = Sys.readdir "directory" in
+Array.iter print_endline files
+```
+In this example, all the file names in the `directory` are printed to the
+console.
+
+### Changing the Current Working Directory
+
+You can change the current working directory using the `Sys.chdir` function.
+
+```ocaml
+Sys.chdir "/path/to/directory"
+```
+In this example, the current working directory is changed to
+`/path/to/directory`.
+
+### Retrieving the Current Working Directory
+
+The current working directory can be retrieved using the `Sys.getcwd` function.
+
+```ocaml
+let cwd = Sys.getcwd ()
+```
+This code stores the current working directory in the `cwd` variable.
 
-  (* flush and close the channel *)
+### Checking If a File or Directory Exists
 
-  (* Read file and display the first line *)
-  let ic = open_in file in
+You can check if a file or directory exists using the `Sys.file_exists`
+function. This function takes a filename or a directory name as an argument and
+returns a boolean indicating whether the file or directory exists.
+
+```ocaml
+if Sys.file_exists "file.txt" then
+  print_endline "File exists."
+else
+  print_endline "File does not exist."
+```
+
+This code checks if `file.txt` exists and prints a message accordingly.
+
+### Checking If a Path is a Directory
+
+The `Sys.is_directory` function is used to check if a path points to a
+directory. It takes a path as an argument and returns a boolean value indicating
+whether the path is a directory.
+
+```ocaml
+if Sys.is_directory "/path/to/directory" then
+  print_endline "It is a directory."
+else
+  print_endline "It is not a directory."
+```
+
+This code checks if `/path/to/directory` is a directory and prints a message
+accordingly.
+
+### Changing File Permissions
+
+OCaml's `Unix` module provides a `chmod` function to change the permissions of a
+file. The function takes a filename and the new permissions (in Unix-style octal
+format) as arguments.
+
+```ocaml
+Unix.chmod "file.txt" 0o644
+```
+
+This code changes the permissions of `file.txt` to read and write for the owner
+and read-only for the group and others.
+
+### Getting File Size
+
+The `Unix.stat` function can be used to get the size of a file. This function
+returns a record with various file attributes, including its size.
+
+```ocaml
+let stats = Unix.stat "file.txt" in
+print_int stats.Unix.st_size
+```
+
+This code gets the size of `file.txt` and prints it.
+
+### Working with File Paths
+
+OCaml's `Filename` module provides several functions to work with file paths.
+
+```ocaml
+let dir = Filename.dirname "/path/to/file.txt" in
+let base = Filename.basename "/path/to/file.txt" in
+let full = Filename.concat dir "new_file.txt" in
+print_endline dir;
+print_endline base;
+print_endline full;
+```
```

Comment:
I think it would be valuable to provide the output of these functions (as comments or below). It might not be immediately clear what are `dir`, `base` and `full` in this context and how functions behave (where they add / remove filepath slashes)